frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1135,6 +1142,11 @@ func dispatchCopy(d *dispatchState, cfg copyConfig) error {
 				a = a.Copy(st, f, dest, opts...)
 			}
 		} else {
+			src, err = system.NormalizePath("/", src, d.platform.OS, false)
```

Comment:
I'm still confused why are doing these path transformations twice. If we already are always converting paths to UNIX in Dockerfile before passing to the LLB then why are we still carrying OS value in LLB and doing extra conversions there.

Reply:
I can remove the calls to `NormalizePath()` from `client/llb` and just use the paths as they come in, if you wish.

The current code in the `master` branch calls [normalizePath()](https://github.com/moby/buildkit/blob/master/client/llb/fileop.go#L777-L794). This code is now incorporated into `system.NormalizePath()`, with additional logic to take into account the input `platform`.

Would you prefer we do away with this and just use `path.Join()` in `toProtoAction()` (example: https://github.com/moby/buildkit/blob/master/client/llb/fileop.go#L340) ?

Reply:
Alternatively, in the client I can just hardcode "linux" as the platform sent into `NormalizePath()` and it would behave just like the current `normalizePath()` in `client/llb`, but I am not sure ignoring the input platform (if set) is worth it. In any case, let me know what you prefer and I will make the change.

Reply:
If you don't need `NormalizePath` calls inside `client/llb` to make windows work then better to remove it.

Reply:
I can revert the change to that file and bring back `normalizePath()` as it is in the `master` branch, because that is somewhat still needed. Although that code is now incorporated in `system.NormalizePath()`.

Will change in the morning. I need to see if anything breaks (with a clear head).

Reply:
I reverted most of the changes to the client. Please take another look.